components/hab/src/main.rs

```diff
@@ -706,6 +754,352 @@ fn sub_pkg_channels(ui: &mut UI, m: &ArgMatches) -> Result<()> {
     command::pkg::channels::start(ui, &url, &ident, token.as_ref().map(String::as_str))
 }
 
+fn sub_svc_set(m: &ArgMatches) -> Result<()> {
+    toggle_verbosity(m);
+    toggle_color(m);
+    let cfg = config::load()?;
+    let sup_addr = sup_addr_from_input(m);
+    let secret_key = ctl_secret_key(&cfg)?;
+    let service_group = ServiceGroup::from_str(m.value_of("SERVICE_GROUP").unwrap())?;
+    let mut ui = ui();
+    let mut validate = protocol::ctl::SvcValidateCfg::new();
+    validate.set_service_group(service_group.clone().into());
+    let mut buf = Vec::with_capacity(protocol::butterfly::MAX_SVC_CFG_SIZE);
+    let cfg_len = match m.value_of("FILE") {
+        Some(f) => {
+            let mut file = File::open(f).unwrap();
+            file.read_to_end(&mut buf).unwrap()
+        }
+        None => io::stdin().read(&mut buf).unwrap(),
+    };
+    if cfg_len > protocol::butterfly::MAX_SVC_CFG_SIZE {
+        ui.fatal(format!(
+            "Configuration too large. Maximum size allowed is {} bytes.",
+            protocol::butterfly::MAX_SVC_CFG_SIZE
+        ))?;
+        process::exit(1);
+    }
+    validate.set_cfg(buf.clone());
+    let cache = default_cache_key_path(Some(&*FS_ROOT));
+    let mut set = protocol::ctl::SvcSetCfg::new();
+    match (service_group.org(), user_param_or_env(&m)) {
+        (Some(_org), Some(username)) => {
+            let user_pair = BoxKeyPair::get_latest_pair_for(username, &cache)?;
+            let service_pair = BoxKeyPair::get_latest_pair_for(&service_group, &cache)?;
+            ui.status(
+                Status::Encrypting,
+                format!(
+                    "TOML as {} for {}",
+                    user_pair.name_with_rev(),
+                    service_pair.name_with_rev()
+                ),
+            )?;
+            set.set_cfg(user_pair.encrypt(&buf, Some(&service_pair))?);
+            set.set_is_encrypted(true);
+        }
+        _ => set.set_cfg(buf.to_vec()),
+    }
+    set.set_service_group(service_group.into());
+    set.set_version(value_t!(m, "VERSION_NUMBER", u64).unwrap());
+    ui.begin(format!(
+        "Setting new configuration version {} for {}",
+        set.get_version(),
+        set.get_service_group(),
+    ))?;
+    ui.status(
+        Status::Creating,
+        format!("service configuration"),
```

Comment:
Feels like this UI message should be something about validating the configuration, rather than creating it.

Reply:
Fixed it